projects/vdk-plugins/vdk-logging-json/src/vdk/plugin/logging_json/logging_json.py

```diff
@@ -46,25 +50,38 @@ def format(self, record):
         return json.dumps(result)
 
 
+@hookimpl
+def vdk_configure(config_builder: ConfigurationBuilder):
+    config_builder.add(
+        key=LOGGING_FORMAT,
+        default_value="TEXT",
+        description="The format in which to structure VDK logs.",
+    )
+
+
 @hookimpl(tryfirst=True)
 def vdk_start(plugin_registry: IPluginRegistry, command_line_args: List):
-    for handler in logging.getLogger().handlers:
-        handler.setFormatter(
-            EcsJsonFormatter(
-                job_name="",
-                attempt_id="",
-                op_id="",
+    if (
+        os.getenv("VDK_LOGGING_FORMAT") == "JSON"
+    ):  # workaround: config isn't initialized yet but logs are being printed
+        for handler in logging.getLogger().handlers:
+            handler.setFormatter(
+                EcsJsonFormatter(
+                    job_name="",
+                    attempt_id="",
+                    op_id="",
+                )
             )
-        )
 
 
 @hookimpl(trylast=True)
 def initialize_job(context: JobContext) -> None:
-    attempt_id = context.core_context.state.get(CommonStoreKeys.ATTEMPT_ID)
-    op_id = context.core_context.state.get(CommonStoreKeys.OP_ID)
-    job_name = context.name
-
-    for handler in logging.getLogger().handlers:
-        handler.setFormatter(
-            EcsJsonFormatter(job_name=job_name, attempt_id=attempt_id, op_id=op_id)
-        )
+    if context.core_context.configuration.get_value(LOGGING_FORMAT) == "JSON":
+        attempt_id = context.core_context.state.get(CommonStoreKeys.ATTEMPT_ID)
+        op_id = context.core_context.state.get(CommonStoreKeys.OP_ID)
+        job_name = context.name
+
+        for handler in logging.getLogger().handlers:
+            handler.setFormatter(
+                EcsJsonFormatter(job_name=job_name, attempt_id=attempt_id, op_id=op_id)
+            )
```

Comment:
Please extract it as method https://refactoring.guru/extract-method which can be used above as well.

Reply:
done